cli/clibase/values.go

```diff
@@ -16,6 +16,49 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
+// Validator is a wrapper around a pflag.Value that allows for validation
+// of the value after or before it has been set.
```

Comment:
I like that this implements `pflag.Value`.

Reply:
Maybe there's a function `Required[T any]() ValidateFunc[T]`

Reply:
> Maybe there's a function `Required[T any]() ValidateFunc[T]`

I was trying, but unfortunately `Validate()` can only validate inputs. If there is no input, `Set()` is never called for this to trigger on.

The only way I could think of making it work is have some post-processing step for the option? It felt easier to make it a flag on the Options and just touch it at the flag parsing.

Reply:
I think it's a common enough validation requirement that making it apart of the Option field is justified.